.NET/Microsoft.Recognizers.Text.DataTypes.DataDrivenTests/TestTimexConvert.cs

```diff
@@ -15,276 +15,314 @@ public void DataTypes_Convert_CompleteDate()
-            Assert.AreEqual("third week of August", TimexConvert.ConvertTimexToString(new TimexProperty("XXXX-08-W03")));
+            Assert.AreEqual(expected, TimexConvert.ConvertTimexToString(new TimexProperty(timex)));
         }
 
         [TestMethod]
-        public void DataTypes_Convert_PartOfTheDay()
+        [DataRow("TDT", "daytime")]
+        [DataRow("TNI", "night")]
+        [DataRow("TMO", "morning")]
+        [DataRow("TAF", "afternoon")]
+        [DataRow("TEV", "evening")]
+        public void DataTypes_Convert_PartOfTheDay(string timex, string expected)
         {
-            Assert.AreEqual("daytime", TimexConvert.ConvertTimexToString(new TimexProperty("TDT")));
-            Assert.AreEqual("night", TimexConvert.ConvertTimexToString(new TimexProperty("TNI")));
-            Assert.AreEqual("morning", TimexConvert.ConvertTimexToString(new TimexProperty("TMO")));
-            Assert.AreEqual("afternoon", TimexConvert.ConvertTimexToString(new TimexProperty("TAF")));
-            Assert.AreEqual("evening", TimexConvert.ConvertTimexToString(new TimexProperty("TEV")));
+            Assert.AreEqual(expected, TimexConvert.ConvertTimexToString(new TimexProperty(timex)));
         }
 
         [TestMethod]
-        public void DataTypes_Convert_FridayEvening()
+        [DataRow("(XXXX-XX-XX,XXXX-XX-XX,P1D)", "1 day")]
+        public void DataTypes_Convert_Period(string timex, string expected)
         {
-            Assert.AreEqual("Friday evening", TimexConvert.ConvertTimexToString(new TimexProperty("XXXX-WXX-5TEV")));
+            Assert.AreEqual(expected, TimexConvert.ConvertTimexToString(new TimexProperty(timex)));
         }
 
         [TestMethod]
-        public void DataTypes_Convert_DateAndPartOfDay()
+        [DataRow("XXXX-WXX-5TEV", "Friday evening")]
+        public void DataTypes_Convert_FridayEvening(string timex, string expected)
         {
-            Assert.AreEqual("7th September 2017 night", TimexConvert.ConvertTimexToString(new TimexProperty("2017-09-07TNI")));
+            Assert.AreEqual(expected, TimexConvert.ConvertTimexToString(new TimexProperty(timex)));
         }
 
         [TestMethod]
-        public void DataTypes_Convert_Last5Minutes()
+        [DataRow("2017-09-07TNI", "7th September 2017 night")]
+        public void DataTypes_Convert_DateAndPartOfDay(string timex, string expected)
         {
-            // date + time + duration
-            var timex = new TimexProperty("(2017-09-08T21:19:29,2017-09-08T21:24:29,PT5M)");
-
-            // TODO
+            Assert.AreEqual(expected, TimexConvert.ConvertTimexToString(new TimexProperty(timex)));
         }
 
         [TestMethod]
-        public void DataTypes_Convert_WednesdayToSaturday()
+        [DataRow("(2017-09-08T21:19:29,2017-09-08T21:24:29,PT5M)", "8th September 2017 9:19:29PM to 9:24:29PM")]
+        public void DataTypes_Convert_Last5Minutes(string timex, string expected)
```

Comment:
What if the dates are different? Do we have a case for that somewhere for it?

Also, this is technically not "last 5 min". It's just a period of 5 min with those parameters.
And the order in the timex is invalid. Start should be smaller than end.

Reply:
I didn't actually do anything with this, I just converted the test to DataRow and added the output that it was missing - the github inline diff is misleading. This is the original code of the test:

```
[TestMethod]
        public void DataTypes_Convert_Last5Minutes()
        {
            // date + time + duration
            var timex = new TimexProperty("(2017-09-08T21:19:29,2017-09-08T21:24:29,PT5M)");

            // TODO
        }
```

I didn't change anything in the minute duration to string conversion, so the output here is the same output that it has always had. I agree that the output doesn't seem to match the output of the test, I just added the output of the original test here so I could see what it was doing.